cookbook/add-tabs.rst

```diff
@@ -0,0 +1,300 @@
+
+
+This tutorial will walk you through the process of adding extra tabs to the user interface of Sulu.
+
+This could be useful in a few situations, a few examples below:
+
+- Saving extra data for pages
+- Saving extra data in webspaces
+- Saving extra data for users
+
+In this example we'll be adding a "Socials" tab to the PageAdmin of Sulu.
+
+.. figure:: ../img/socials-tab.png
+
+Create the form for this view
+-----------------------------
+You have to create the form that is rendered in this view. For this we create an XML file like your used to.
+Create your form: ``config/forms/page_socials.xml``. Notice how we use slashes in the names of the properties, this will be of later use.
+
+.. code-block:: xml
+
+    <?xml version="1.0" ?>
+    <form xmlns="http://schemas.sulu.io/template/template"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/form-1.0.xsd"
+    >
+        <key>page_socials</key>
+
+        <properties>
+            <section name="twitter">
+                <meta>
+                    <title>Twitter</title>
+                </meta>
+                <properties>
+                    <property name="ext/social/twitter_title" type="text_line">
+                        <meta>
+                            <title>Twitter title</title>
+                        </meta>
+                    </property>
+                    <property name="ext/social/twitter_description" type="text_line">
+                        <meta>
+                            <title>Twitter description</title>
+                        </meta>
+                    </property>
+                    <property name="ext/social/twitter_image" type="single_media_selection">
+                        <meta>
+                            <title>Twitter image</title>
+                        </meta>
+                        <params>
+                            <param name="types" value="image"/>
+                            <param name="formats" type="collection">
+                                <param name="og_image" />
+                            </param>
+                        </params>
+                    </property>
+                </properties>
+            </section>
+        </properties>
+    </form>
+
+
+
+Registering your own admin class
+--------------------------------
+
+We'll create a class for example ``src/Admin/PageAdminExtension`` which extends ``Sulu\Bundle\AdminBundle\Admin``.
+
+In the admin we need to implement configureViews. Also for our example we use the `webspace_manager` in combination with `security_checker` to check if the logged in user has permission to edit pages.
+
+Also important, notice how we set the parent to be `PageAdmin::EDIT_FORM_VIEW`. This adds our view to the `PageAdmin::EDIT_FORM_VIEW` view.
+
+The setFormKey is a reference to the <key> in the created form from the above section.
```

Comment:
"`setFormKey` takes a string reference, which should be the same as the `key` tag in the above form XML."